api/v1alpha1/serviceintentions_webhook.go

```diff
@@ -85,10 +90,33 @@ func (v *ServiceIntentionsWebhook) Handle(ctx context.Context, req admission.Req
 	if err := svcIntentions.Validate(); err != nil {
 		return admission.Errored(http.StatusBadRequest, err)
 	}
-	return admission.Allowed(fmt.Sprintf("valid %s request", svcIntentions.KubeKind()))
+	// We always return an admission.Patched() response, even if there are no patches, since
+	// admission.Patched() with no patches is equal to admission.Allowed() under
+	// the hood.
+	return admission.Patched(fmt.Sprintf("valid %s request", svcIntentions.KubeKind()), defaultingPatches...)
 }
 
 func (v *ServiceIntentionsWebhook) InjectDecoder(d *admission.Decoder) error {
 	v.decoder = d
 	return nil
 }
+
+// defaultingPatches returns the patches needed to set fields to their
+// defaults.
+func (v *ServiceIntentionsWebhook) defaultingPatches(err error, svcIntentions *ServiceIntentions) ([]jsonpatch.Operation, error) {
```

Comment:
This only returns patches in consul ent but I didn't short-circuit in case we add defaulting logic for oss later.

Reply:
This comment just stopped me from mentioning we should short circuit it 😛